lib/core/test/unit/Cardano/Wallet/Primitive/CoinSelection/MA/RoundRobinSpec.hs

```diff
@@ -257,30 +264,71 @@ prop_runSelection_UTxO_notEnough (Small index) = monadicIO $ do
     balanceRequested = adjustAllQuantities (* 2) balanceAvailable
 
 prop_runSelection_UTxO_exactlyEnough
-    :: Small UTxOIndex -> Property
+    :: Small UTxOIndex
+    -> Property
 prop_runSelection_UTxO_exactlyEnough (Small index) = monadicIO $ do
     SelectionState {selected, leftover} <-
-        run $ runSelection index balanceRequested
+        run $ runSelection Nothing index balanceRequested
     let balanceSelected = view #balance selected
     let balanceLeftover = view #balance leftover
     assert $ balanceSelected == balanceRequested
     assert $ balanceLeftover == TokenBundle.empty
   where
     balanceRequested = view #balance index
 
+prop_runSelection_UTxO_extraSourceUsed
+    :: Maybe Coin
+    -> Small UTxOIndex
+    -> Property
+prop_runSelection_UTxO_extraSourceUsed extraSource (Small index) =
+    monadicIO $ case almostEverything of
+        Nothing ->
+            assert True
+        Just balanceRequested | balanceRequested == TokenBundle.empty -> do
+            SelectionState {selected,leftover} <-
+                run $ runSelection extraSource index balanceRequested
+            let balanceSelected = view #balance selected
+            let balanceLeftover = view #balance leftover
+            assert $ balanceLeftover == view #balance index
+            assert $ balanceSelected == TokenBundle.empty
+        Just balanceRequested -> do
+            monitor (cover 80 True "sometimes there are Ada")
```

Comment:
I suspect that if this case is never matched, the coverage check will not cause this property to fail, as it will never be reached.

Perhaps it's better to write a coverage check at the top-level that looks something like this:

```hs
prop_runSelection_UTxO_extraSourceUsed extraSource (Small index) =
    checkCoverage $
    monitor $ cover 80 (isJust balanceRequested)
        "sometimes there are Ada"
    monadicIO $ case almostEverything of
    
```